src/main/kotlin/org/rust/ide/newProject/RsProjectTemplate.kt

```diff
@@ -14,12 +14,18 @@ sealed class RsProjectTemplate(val name: String, val isBinary: Boolean) {
     fun validateProjectName(crateName: String): String? = RsPackageNameValidator.validate(crateName, isBinary)
 }
 
-class RsGenericTemplate(name: String, isBinary: Boolean) : RsProjectTemplate(name, isBinary) {
+sealed class RsGenericTemplate(name: String, isBinary: Boolean) : RsProjectTemplate(name, isBinary) {
     override val icon: Icon = RsIcons.RUST
 }
 
-class RsCustomTemplate(name: String, val link: String, isBinary: Boolean = true) : RsProjectTemplate(name, isBinary) {
+sealed class RsCustomTemplate(name: String, val link: String, isBinary: Boolean = true) : RsProjectTemplate(name, isBinary) {
```

Comment:
Let's move objects inside and get rid of `isBinary` parameter and assume that `RsCustomTemplate` is always **not binary** to avoid false-negative name validation errors